sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py

```diff
@@ -142,6 +141,25 @@ def _cloud_event_to_generated(cloud_event, **kwargs):
         **kwargs
     )
 
+def _from_cncf_events(event):
+    """This takes in a CNCF cloudevent and returns a dictionary.
+
+    :raises: AttributeError when a dictionary is passed/ when data attribute is not
+     present in the event.
+    """
+    try:
+        from cloudevents.http import to_json
+        return json.loads(to_json(event))
+    except AttributeError:
+        # means this is not a CNCF event
+        return event
+    except ImportError:
+        # this means it was not meant to be a cloud event
+        return None
+    except: # pylint: disable=bare-except
+        raise ValueError('Failed to deserialize the event.')
```

Comment:
Is that a failed to serialize actually? Since the method is about serializing CloudEvent to send right?
Could we make the exception more precise about what people should do to react to that error?

Reply:
pointing to the library and improved the error message